server/hooks.go

```diff
@@ -17,26 +17,65 @@ func (p *Plugin) UserHasJoinedTeam(c *plugin.Context, teamMember *model.TeamMemb
 		return
 	}
 
-	for _, message := range p.getWelcomeMessages() {
-		if message.TeamName == data.Team.Name {
-			go p.processWelcomeMessage(*data, *message)
+	key := fmt.Sprintf("%s%s", welcomebotTeamWelcomeKey, teamMember.TeamId)
+	teamMessage, appErr := p.API.KVGet(key)
+	if appErr != nil {
+		mlog.Error(
+			"error occurred while retrieving the welcome message",
+			mlog.String("teamId", teamMember.TeamId),
+			mlog.Err(appErr),
+		)
+		return
+	}
+
+	if teamMessage == nil {
+		// No dynamic welcome message for the given team, so we check if one as been set in the config.json
+		for _, message := range p.getWelcomeMessages() {
+			if message.TeamName == data.Team.Name {
+				go p.processWelcomeMessage(*data, *message)
+			}
 		}
+		return
+	}
+
+	// We send a DM and an opportunistic ephemeral message to the channel. See
+	// the discussion at the link below for more details:
+	// https://github.com/mattermost/mattermost-plugin-welcomebot/pull/31#issuecomment-611691023
+	teamAddMessage := "# Welcome to the " + data.Team.DisplayName + " team!! \n\n"
```

Comment:
Why is this default message needed?

Reply:
This change I believe was added to give some kind of header to the welcome bot messages. However, if this ultimately feels like we are stripping freedoms to the welcome message, it can be thrown out.